docker/Dockerfile

```diff
@@ -12,18 +12,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Create basic runtime environment base on Ubuntu 22.04 (jammy)
 # Define build arguments
-# Create and activate runtime virtual environment
-FROM ubuntu:jammy AS base-runtime
 ARG CUDA_ARCH=AMPERE80
 ARG AMD_ARCH=AMD_GFX90A
 ARG CUDA_INSTALLER=https://developer.download.nvidia.com/compute/cuda/12.2.2/local_installers/cuda_12.2.2_535.104.05_linux.run
 ARG DEBIAN_FRONTEND=noninteractive
 ARG GCC_VERSION=11
-ARG LIGHTNING_VERSION=v0.36.0
-ARG PENNYLANE_VERSION=v0.36.0
+ARG LIGHTNING_VERSION=master
+ARG PENNYLANE_VERSION=master
 ARG ROCM_INSTALLER=https://repo.radeon.com/amdgpu-install/5.7/ubuntu/jammy/amdgpu-install_5.7.50700-1_all.deb
+
+# Create basic runtime environment base on Ubuntu 22.04 (jammy)
+# Create and activate runtime virtual environment
+FROM ubuntu:jammy AS base-runtime
+ARG CUDA_ARCH
+ARG AMD_ARCH
+ARG CUDA_INSTALLER
+ARG DEBIAN_FRONTEND
+ARG GCC_VERSION
```

Comment:
Any reason to redefine  `CUDA_ARCH`, `AMD_ARCH`... here?

Reply:
The defaults ARGs are defined at the top and ARGs must be imported in each stage that is not created from a previous one. Now that you mention it, I think most of them can go directly in `base-runtime`, but `PENNYLANE_VERSION` can't, for example. I'll see whether I can make it neater.

Reply:
Works, thanks @multiphaseCFD .